### PR TITLE
Easy Setup

### PR DESCRIPTION
**Pull request by afiune**
As a user I would like to have an easier way to setup my Delivery Cluster
locally so I can start getting familiarize with it and then customize it
like a master. (Issue https://github.com/opscode-cookbooks/delivery-cluster/issues/113)

New cool `rake` task called `generate_env[name]` that will help you
generate your environment files.

Using the defaults it will generate this environment file:
```
$ rake setup:generate_env[test]

Gathering Cluster Information

Provide the following information to generate your environment.

Global Attributes
Cluster ID [test]:
Driver Name [ssh]:

Driver Information [ssh]
SSH Username [vagrant]:
Key File [/Users/salimafiune/.vagrant.d/insecure_private_key]:

Chef Server
Organization Name [test]:
Use existing chef-server? [no]:
Host [33.33.33.10]:

Delivery Server
Package Version [latest]:
Enterprise Name [test]:
Use chef artifactory? [no]:
License File [/Users/salimafiune/github/delivery-cluster/delivery.license]:
Host [33.33.33.11]:

Analytics Server
Enable Analytics? [no]:

Supermarket Server                                                                                                                                                           Chef-12.3.0 Ruby 2.1.5p273

Build Nodes
Number of Build Nodes [1]:
Host for Build Node 1 [33.33.33.14]:

Rendering Environment => environments/test.json

{
  "name": "test",
  "description": "Delivery Cluster Environment",
  "json_class": "Chef::Environment",
  "chef_type": "environment",
  "override_attributes": {
    "delivery-cluster": {
      "id": "test",
      "driver": "ssh",
      "ssh": {
        "ssh_username": "vagrant",
        "key_file": "/Users/salimafiune/.vagrant.d/insecure_private_key"
      },
      "chef-server": {
        "organization": "test",
        "existing": false,
        "host": "33.33.33.10"
      },
      "delivery": {
        "version": "latest",
        "enterprise": "test",
        "artifactory": false,
        "license_file": "/Users/salimafiune/github/delivery-cluster/delivery.license",
        "host": "33.33.33.11"
      },
      "builders": {
        "count": "1",
        "1": {
          "host": "33.33.33.14"
        }
      }
    }
  }
}

Export your new environment by executing:
  # export CHEF_ENV=test
```

cc/ @opscode-cookbooks/delivery

### CODE REVIEW

**Comment by afiune** [comment]
Gist with the full automation: https://gist.github.com/afiune/475c559b842687f27e47

File: README.md [hunk 1]
```diff
@@ -68,6 +69,92 @@ To switch your environment run:
...
+not have a license key, you can request one from your CHEF account
+representative.**
+
+You will need to have the `delivery.license` file present on your provisioner
+node or local workstation.
+
+#### 2) Provisioning infrastructure [SSH/Kitchen]
+
+You can provision your infrastructure on your prefered provider. We will use
+[KichenCI](http://kitchen.ci/) for the easy setup so you can get familiarize.
```
Comment (jonsmorrow) [inline README.md]: s/familiarize/familiar

**Comment by tduffield** [comment]
Some thoughts:
* My zsh install doesn't like the brackets ([]) without escaping them. I would recommending either making a small note about the potential necessity of escaping brackets or (my preference) having them enter the name in the interaction.
* If you don't have a license file when you start the process it won't let you continue. Given that questions location in the questionnaire I worry that forcing them to pause and/or quite to deal with that issue right away could be unfriendly. I'd much rather see, if you can't find the license file, that you ask them to confirm the location or let them know that they'll need that file in that location before they can build the cluster but let them proceed with the questions. 

Other than those things it looks awesome.

**Comment by jonsmorrow** [comment]
Agree with tom's comments.

I am a little worried we are coding the shape of this file, ie the specific attributes, in a lot of places. This means if we need to add an additional field it must be done all over the place. 

Overall though this is kick ass.

**Comment by afiune** [comment]
@jonsmorrow @tduffield what about this ¿? :smile:

**Comment by tduffield** [comment]
I think its a good middle ground. The user still couldn't create the environment without a license locally which I think could cause issues but I say we deal with that if it ever comes up.

**Comment by afiune** [comment]
@tduffield is that a +1 ? or should I change something else? :smile:

**Comment by tduffield** [comment]
Unless anyone else has something else to add I'm :+1:

**Comment by jonsmorrow** [comment]
:+1:

**Comment by binamov** [comment]
this works :+1:

**Pull request merged**